cpp/models/abm/location.cpp

```diff
@@ -39,30 +39,31 @@ Location::Location(LocationType type, uint32_t index, uint32_t num_cells)
 {
 }
 
-InfectionState Location::interact(const Person& person, TimeSpan dt,
-                                  const GlobalInfectionParameters& global_params) const
+Infection* Location::interact(const Person& person, const TimePoint& t, const TimeSpan dt,
```

Comment:
Why using a normal pointer here? Can we avoid that? We could also a return a COD which is only "Susceptible"? Just a quick shot, dont know if this is the best idea.

Reply:
Reflecting on this, a person should probably have a unique_ptr to an infection anyway. Since a new infection is created here, the unique_ptr could then be passed as return of this function.

Reply:
Outdated. The Location::interact function now immediately creates infections and applies them to the persons.